korman/exporter/camera.py

```diff
@@ -206,6 +206,9 @@ def _export_rail_camera(self, so, bo, props):
             f1, f2 = fcurve.evaluate(begin), fcurve.evaluate(end)
             if abs(f1 - f2) > 0.001:
                 break
+            # to avoid single/duplicate keyframe client crash - Doobes
+            elif f1 == f2:
```

Comment:
This will probably never get triggered. If I'm remembering correctly, it is legal for the first and last keyframe to be in roughly the same place, which triggers a special "wrapping" kind of rail camera. The if statement is to give some wiggle room for engaging that mode. I think what we need to do is make sure the final exported `pos_ctrl` has more than a single keyframe. We should probably be testing that deeper inside the animation code to make sure no animations have a single keyframe. For now, though, we can do something like:

```python
if any((len(i.keys == 1) for i in (pos_ctrl.X, pos_ctrl.Y, pos_ctrl.Z) if i is not None)):
    raise ExportError("'{}': Rail Camera must have more than one keyframe", bo.name)
```

before line 201

Reply:
If it's a wrapping camera, wouldn't a circle camera make more sense? Unless it's not a uniform circle path, I suppose.

Note that this *did* trigger in my testing of it, both with single keyframe and two keyframes in the same position. I suppose if the latter doesn't crash the client, then your code is probably much more the scalpel to my sledgehammer. 😉

Reply:
It might be a square or some other odd type path :wink:

Reply:
> 
> 
> It might be a square or some other odd type path 😉

Good point! We artists tend to like to do crazy stuff.

Reply:
Whoops, let's try:
```python
if any((len(i.keys) == 1 for i in (pos_ctrl.X, pos_ctrl.Y, pos_ctrl.Z) if i is not None)):
    raise ExportError("'{}': Rail Camera must have more than one keyframe", bo.name)
```

Reply:
Seems to work! I tried exporting with three keyframes (passed), one keyframe for all curves (stopped) and three keyframes for only the y curve (passed)